OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1

```diff
@@ -51,11 +51,15 @@ function Get-MachineFromOctopusServer
 
         [Parameter()]
         [System.String]
-        $Instance
+        $Instance,
+
+        [Parameter()]
+        [System.String]
+        $SpaceId = "Spaces-1"
     )
 
     # Get all the machines form Octopus
-    $machines = Get-APIResult -ServerUrl $ServerUrl -APIKey $APIKey -API "/machines/all"
+    $machines = Get-APIResult -ServerUrl $ServerUrl -APIKey $APIKey -API "/$SpaceId/machines/all"
```

Comment:
This wont work on a pre-spaces Octopus server

Reply:
Updated code to include space id in url only if space was provided.